### DISCUSSION TITLE
[VTK] How should I enable a module

### DISCUSSION THREAD

**Discussion by larshg** [Homebrew/discussions#9] [Everyday usage]
So I just add this to the formula? Or can it be added as an optional like in VCPKG?

DVTK_MODULE_ENABLE_VTK_RenderingContextOpenGL2

Currently its not found as seen on this CI:
https://dev.azure.com/PointCloudLibrary/pcl/_build/results?buildId=17496&view=logs&j=8042da28-3549-5cef-c93d-1a000d12f42a&t=fece401d-112c-54c6-d591-56921a9f56a6

**Comment by SMillerDev** [comment]
Is it a module that's on by default? Does upstream recommend enabling it? Is it useful to most homebrew users?

**Comment by larshg** [comment]
I doesn't seem to be on by default. I can hardly find anything about recommending it, but some people have been used to it just being there(from 8.2):
https://discourse.vtk.org/t/9-0-0-rc1-contextopengl2-not-built/2933

It enables homebrew users to render 2D images using opengl, instead of maybe just software rendering.

vtkcontextdevice2d vs vtkopenglcontextdevice2d.

**Reply by SMillerDev** [reply]
OpenGL is deprecated on macOS. I'm not sure if we want to enable that for everyone.

**Reply by SMillerDev** [reply]
https://developer.apple.com/library/archive/documentation/GraphicsImaging/Conceptual/OpenGL-MacProgGuide/opengl_intro/opengl_intro.html for reference

**Comment by larshg** [comment]
Yeah, I have heard that as well. Deprecated, but not yet gone :) And there are already being compiled some OpenGL dependent sublibraries in VTK by default.

But hopefully VTK is working in some direction of supporting another rendering engine at some point.
